src/components/ComputeModal.js

```diff
@@ -1012,6 +1082,33 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
     ])
   }
 
+  const renderDifferentLocationWarning = () => {
+    return div({ style: { ...computeStyles.drawerContent, ...computeStyles.warningView } }, [
+      h(TitleBar, {
+        id: titleId,
+        hideCloseButton: isAnalysisMode,
+        style: computeStyles.titleBar,
+        title: h(WarningTitle, ['Compute location differs from workspace bucket location']),
+        onDismiss,
+        onPrevious: () => setViewMode(undefined)
+      }),
+      div({ style: { lineHeight: 1.5 } }, [
+        p([
+          'This cloud environment will be created in the region ', computeConfig.computeRegion.toLowerCase(), '. ',
+          'Copying data from your workspace bucket in ', bucketLocation.toLowerCase(), ' may incur network egress charges.'
```

Comment:
We usually use string extrapolation to embed code, which to me is a bit easier to read, a bit harder to mess up on an IDE :)
```suggestion
          `This cloud environment will be created in the region ${computeConfig.computeRegion.toLowerCase()}. `,
          `Copying data from your workspace bucket in ${bucketLocation.toLowerCase()} may incur network egress charges.`
```